### PR TITLE
Add port F for Arduino Leonardo

### PR DESCRIPTION
**Pull request by cecton**
Related to https://github.com/Rahix/avr-hal/pull/62#issuecomment-689419628

### CODE REVIEW

File: boards/arduino-leonardo/src/pins.rs [hunk 1]
```diff
@@ -10,12 +10,23 @@ avr_hal_generic::impl_board_pins! {
...
+        /// `A1`
+        pub a1: portf::pf6::PF6,
+        /// `A2`
+        pub a2: portf::pf5::PF5,
+        /// `A3`
+        pub a3: portf::pf4::PF4,
+        /// `A4`
+        pub a4: portf::pf1::PF1,
+        /// `A5`
+        pub a5: portf::pf0::PF0,
```
Comment (Rahix) [inline boards/arduino-leonardo/src/pins.rs:29]: Please also document the alternate functions for these pins.  I'd also move this block to the end, so the digital pins are first in the documentation.

```suggestion
        /// `A0`
        ///
        /// * `ADC7` channel
        pub a0: portf::pf7::PF7,
        /// `A1`
        ///
        /// * `ADC6` channel
        pub a1: portf::pf6::PF6,
        /// `A2`
        ///
        /// * `ADC5` channel
        pub a2: portf::pf5::PF5,
        /// `A3`
        ///
        /// * `ADC4` channel
        pub a3: portf::pf4::PF4,
        /// `A4`
        ///
        /// * `ADC1` channel
        pub a4: portf::pf1::PF1,
        /// `A5`
        ///
        /// * `ADC0` channel
        pub a5: portf::pf0::PF0,
```

File: boards/sparkfun-pro-micro/examples/pro-micro-adc.rs [hunk 2]
```diff
@@ -19,7 +20,7 @@ fn main() -> ! {
 
     ufmt::uwriteln!(&mut serial, "Reading analog inputs ...\r").void_unwrap();
 
-    let mut adc = sparkfun_pro_micro::adc::Adc::new(dp.ADC, Default::default());
+    let mut adc = adc::Adc::new(dp.ADC, Default::default());
```
Comment (Rahix) [inline boards/sparkfun-pro-micro/examples/pro-micro-adc.rs:23]: Let's keep those changes out of this MR and do them later (if at all) once we have decided what to do with the crate.

**Review by Rahix — CHANGES_REQUESTED** [review]
Thanks a lot!  Minor nitpick in the comment below

File: boards/sparkfun-pro-micro/examples/pro-micro-adc.rs (hunk 2 above)
Comment (cecton) [reply boards/sparkfun-pro-micro/examples/pro-micro-adc.rs]: done

File: boards/arduino-leonardo/src/pins.rs (hunk 1 above)
Comment (cecton) [reply boards/arduino-leonardo/src/pins.rs]: done & done (ADCx are the alternate functions?)

Comment (Rahix) [reply boards/arduino-leonardo/src/pins.rs]: Correct, by default the pins are Digital I/O, and they can be reconfigured as ADC channels.

**Review by Rahix — APPROVED** [review]

**Pull request merged**